File: sample_code.m
clc; clear all; format short

% Define wavelet parameters
wavelet_type = 'sym3';
wavelet_level = 5;

% Generate input data
N = 9;
signal = 0:N-1;

% Decompose input signal
[C, L] = wavedec(signal, wavelet_level, wavelet_type);

% Reconstruct all the coefficients
for n=1:max(1, length(L)-2), disp(n)
    D = wrcoef('d', C, L, 'sym3', n);
    disp(D)
end

File: sample_code.py
# -*- coding: utf-8 -*-
import pywt
from wrcoef import wavedec, wrcoef


# Define wavelet parameters
wavelet_type = 'sym3'
wavelet_level = 5

# Generate input data
N = 9
signal = range(N)

# Define wavelet
w = pywt.Wavelet(wavelet_type)
# print([w.dec_lo, w.dec_hi, w.rec_lo, w.rec_hi])

# Decompose input signal
C, L = wavedec(signal, wavelet=w, level=wavelet_level)

# Reconstruct all the coefficients
for n in range(len(L)-2):
    D = wrcoef(C, L, wavelet=w, level=n+1)
    print(n+1, D)
